### DISCUSSION TITLE
Should use clearText instead of rawData in InflaterInputBuffer.ReadLeByte()?

### DISCUSSION THREAD

**Discussion by uscq** [icsharpcode/SharpZipLib#806] [Q&A]
In InflaterInputBuffer, when specifies the CryptoTransform by call `zip.cryptoTransform = new MyCustomTransform();`, the raw data (`rawData`) may has been transformed to `clearText`.
Why still use the `clearText` to read the data in `InflaterInputBuffer.ReadLeByte()` instead use `rawData`.

code at https://github.com/icsharpcode/SharpZipLib/blob/cdd0931e3c476faaf829963b2567619431c871bb/src/ICSharpCode.SharpZipLib/Zip/Compression/Streams/InflaterInputStream.cs#L239

**Comment by piksel** [comment]
I think what you are asking is about why the crypto transform is not performed on the data read from `ReadLeByte()` et al.
The reason is basically that InflaterInputStream is used for two purposes, ZIP and GZip, and only the latter of them contains a header in the "deflated" data. The `ReadLe*()` methods are used to read said header, which should not be encrypted. That being said, GZip doesn't support encryption at all, so it shouldn't matter if the transform is used or not.
